pkg/openshift/deploymentconfig_test.go

```diff
@@ -0,0 +1,208 @@
+package openshift_test
```

Comment:
This is pretty much identical to k8s/deployment_test.go, same with the impl for that matter. 'maybe' worth creating a single impl that takes a 'type' handle of some sort as input? @bartoszmajsak wdyt?

Reply:
From one side that would ensure we cover paths for both `Deployment` and `DeploymentConfig`. I don't know by heart if there are any substantial differences between those objects we would need to cover separately (e.g. triggers?), so not sure how much we will diverge in the implementation yet. This actually raises a question of why we don't handle both types in one set of components.

As for the implementation - DataTable tests might be a way to have parametrized input for those test cases, but I am not sure that is a fit here. We are looking for suite-level input, not really per-each functionality.

Reply:
Not 100% sure how much they might diverge either, which is why I'm a little eerie to merge them. Certainly, there is an overlap on the general 'blackbox' level. 

I've got an idea, let me see if it works :)

Reply:
yea... terrible idea. two sets there is :)